### PR TITLE
refactor: rename account registry, namespace 

### PR DESCRIPTION
**Pull request by varunsrin**
Updating the spec to reflect the changes made to contract names as part of https://github.com/farcasterxyz/contracts/pull/30

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -53,17 +53,17 @@ User data is then cryptographically signed by the identity and stored off-chain
 
 A Farcaster account is similar to an account on pseudonymous social networks like Twitter or Reddit. Individuals can operate several accounts simultaneously, like a real-name account and a company account.
 
-An Ethereum address can mint a new account from the _AccountRegistry_, which issues an account number. This address is known as the `custody address`, and it can sign messages on behalf of the account. Accounts can be enriched by adding a profile picture, display name, biography, and verified usernames like alice.eth, which are set off-chain with signed messages.
+Each account has by a unique number associated with it called a Farcaster ID or a `fid`. A fid can be obtained by calling the *Farcaster ID Registry* or *FIR* from an Ethereum address. This address is known as the `custody address` and can sign off-chain and on-chain messages on behalf of the account. Users can optionally acquire a Farcaster Name or `fname` from the *Farcaster Name Registry* or *FNR* which issues it a unique name like `@alice`.
```
Comment (pfletcherhill) [inline README.md:56]: Nit: you go back and forth (see line 81) about _an_ fid or _a_ fid. I'd pick _an_ fid.

Comment (pfletcherhill) [inline README.md:56]: Another nit: this paragraph has a lot of "...big long name or acronym..." phrases. How about "...big long name (acronym)..." instead?

File: README.md [hunk 2]
```diff
@@ -53,17 +53,17 @@ User data is then cryptographically signed by the identity and stored off-chain
...
-Signed Messages are **tamper-proof** and **self-authenticating** objects that are signed by an account.
+Signed Messages are **tamper-proof** and **self-authenticating** objects that are signed by an fid. They represent user actions like making a public post, reacting to something or adding metadata to their account like a username.
 
 ```ts
 type SignedMessage = {
   message: {
-    body: any;
-    account: number;
+    body: any; 
+    fid: number;
```
Comment (pfletcherhill) [inline README.md:66]: Nice, I like that

File: README.md [hunk 3]
```diff
@@ -76,53 +76,54 @@ type SignedMessage = {
...
 
-The identity registry ensures secure and decentralized ownership of user accounts. Users must be able to set up a new account without any third-party approval. It should also be reasonably quick and cheap, which we define as being able to sign up in under a minute and for less than $10. The system should also ensure that most accounts are trustworthy and operated by honest, active users.
+The identity system ensures secure and decentralized ownership of user accounts. Users must be able to set up a new account without any third-party approval. It should also be reasonably quick and cheap, which we define as being able to sign up in under a minute and for less than $10. The system should also ensure that most accounts are trustworthy and operated by honest, active users.
 
-Farcaster's Identity Registry comprises two smart contracts - an **AccountRegistry**, which issues new accounts, and a **NamespaceRegistry** which issues usernames for accounts.
+Farcaster's Identity System is separated into two on-chain contracts - a **Farcaster ID Registry (FIR)**, which issues new id numbers called `fids`, and a **Farcaster Name Registry (FNR)** which issues new usernames called `fnames`.
```
Comment (pfletcherhill) [inline README.md:107]: Nice and straightforward

File: README.md [hunk 4]
```diff
@@ -76,53 +76,54 @@ type SignedMessage = {
...
 
-## 3.1 Account Registry
 
-An account is created by calling `register` on the AccountRegistry from a user-controlled Ethereum address. The contract issues a new account number and links it to this address. Accounts can be transferred between addresses, though the AccountRegistry ensures that an address owns no more than one account at a time.
+## 3.1 Farcaster ID Registry
```
Comment (pfletcherhill) [inline README.md:110]: I'd make the heading "3.1 Farcaster ID Registry (FIR)", because you refer to it by FIR in most places

File: README.md [hunk 5]
```diff
@@ -76,53 +76,54 @@ type SignedMessage = {
...
 
-Account numbers start at 0 and are incremented by one every time a new account is registered, which is a gas-efficient way to ensure unique account numbers. Numbers are represented internally with a uint256 and have a practically infinite supply since they can be incremented to ~ 10^77.
+An account is created by calling `register` on the ID Registry from a user-controlled Ethereum address. The contract issues a new fid to this address. The fid can be transferred between addresses, though the contract ensures that an address owns only one account at a time.
 
-## 3.2 Namespace Registry
+Fids start at 0 and are incremented by one every time a new registration happens, which is a gas-efficient way to ensure unique account numbers. A fid is represented with a uint256 which guarantees a practically infinite supply since it can be incremented to ~ 10^77.
 
-A Farcaster Name like `@alice` can be minted from the NamespaceRegistry and used on the protocol.
+## 3.2 Farcaster Name Registry
```
Comment (pfletcherhill) [inline README.md:116]: Add "FNR" here too

File: README.md [hunk 6]
```diff
@@ -76,53 +76,54 @@ type SignedMessage = {
...
+Farcaster's Identity System is separated into two on-chain contracts - a **Farcaster ID Registry (FIR)**, which issues new id numbers called `fids`, and a **Farcaster Name Registry (FNR)** which issues new usernames called `fnames`.
 
-## 3.1 Account Registry
 
-An account is created by calling `register` on the AccountRegistry from a user-controlled Ethereum address. The contract issues a new account number and links it to this address. Accounts can be transferred between addresses, though the AccountRegistry ensures that an address owns no more than one account at a time.
+## 3.1 Farcaster ID Registry
 
-Account numbers start at 0 and are incremented by one every time a new account is registered, which is a gas-efficient way to ensure unique account numbers. Numbers are represented internally with a uint256 and have a practically infinite supply since they can be incremented to ~ 10^77.
+An account is created by calling `register` on the ID Registry from a user-controlled Ethereum address. The contract issues a new fid to this address. The fid can be transferred between addresses, though the contract ensures that an address owns only one account at a time.
```
Comment (pfletcherhill) [inline README.md:112]: "...owns only one fid at a time." ?

File: README.md [hunk 7]
```diff
@@ -76,53 +76,54 @@ type SignedMessage = {
...
 
-## 3.2 Namespace Registry
+Fids start at 0 and are incremented by one every time a new registration happens, which is a gas-efficient way to ensure unique account numbers. A fid is represented with a uint256 which guarantees a practically infinite supply since it can be incremented to ~ 10^77.
```
Comment (pfletcherhill) [inline README.md:114]: Nit: maybe use "Farcaster ID" when starting a sentence? In my mind, "Fid" sounds like "fid" and "fid" sounds like "F I D." (Love that that sentence is recorded in github now.)

File: README.md [hunk 8]
```diff
@@ -76,53 +76,54 @@ type SignedMessage = {
...
+Fnames are ERC-721 tokens that are fully composable with the NFT ecosystem. While users can use ENS names with the protocol, Fnames have some properties that make them more practical. Fnames are cheaper to mint and are [recoverable](#recovery) if the address holding them is lost. They are also less vulnerable to [homoglyph attacks](https://en.wikipedia.org/wiki/IDN_homograph_attack) and more straightforward to display because of the restricted length and character set.
 
-Names can be minted freely but are subject to two policies enforced by governance:
+The FNR charges a yearly fee of 0.01 ETH for owning an fname at the beginning of the year on  Farcaster Day (August 1st). Users who do not renew their usernames by this day have a 30-day grace period after which their ownership over the name expires. During mint, the fee is pro-rated for the year ending August 1st. Fees are collected by the Farcaster Treasury and are used to support protocol development.
```
Comment (pfletcherhill) [inline README.md:124]: There's an extra space in between "on" and "Farcaster Day"

File: README.md [hunk 9]
```diff
@@ -132,7 +133,7 @@ Names that expire or are subject to a claim are moved back into the Farcaster Tr
 
 ## 3.3 Recovery
 
-Names and Addresses are recoverable if the user loses the keys to the address holding them. Both contracts implement a time-delayed recovery system that allows a **recovery address** to request a transfer to a new address. If the custody address does not cancel the transfer within three days, the recovery address can complete the transfer.
+Fids and fnames are recoverable if the user loses the keys to the address holding them. Both contracts implement a time-delayed recovery system that allows a **recovery address** to request a transfer to a new address. If the custody address does not cancel the transfer within three days, the recovery address can complete the transfer.
```
Comment (pfletcherhill) [inline README.md:136]: "Farcaster IDs and Names"

File: README.md [hunk 10]
```diff
@@ -173,11 +174,12 @@ Timestamps are compared as numbers, and hashes are compared as strings. Since st
...
 All messages must pass the following validations in addition to specific validations for the message type:
 
-1. `message.timestamp` is not more than 1 hour ahead of system time.
-2. `message.account` must be a known account number in the AccountRegistry.
-3. `signerPubKey` should be a valid [Root Signer or Delegate Signer](#45-signer-authorizations) for `message.account`
-4. `hashFn(serializeFn(message))` must match `envelope.hash`, where hashFn is a Blake2B function and serializeFn performs JSON canonicalization.
-5. `EdDSA_signature_verify(envelope.hash, envelope.signerPubKey, envelope.signature)` should pass.
+1. `message.timestamp` is not more than 1 hour ahead of system time. 
+2. `message.id` must be a known fid number in the FIR.
+3. `signerPubKey` should be a valid [Root Signer or Delegate Signer](#45-signer-authorizations) for `message.id`
```
Comment (pfletcherhill) [inline README.md:179]: `message.fid` here and the line above, no?

File: README.md [hunk 11]
```diff
@@ -173,11 +174,12 @@ Timestamps are compared as numbers, and hashes are compared as strings. Since st
...
+4. `hashFn(serializeFn(message))` must match `envelope.hash`, where hashFn is a Blake2B function and serializeFn performs JSON canonicalization. 
+5.  `EdDSA_signature_verify(envelope.hash, envelope.signerPubKey, envelope.signature)` should pass.
```
Comment (pfletcherhill) [inline README.md:181]: Weird extra space after the number. Would running prettier on the file fix little things like this?

File: README.md [hunk 12]
```diff
@@ -419,11 +422,11 @@ _This section is still under development._
 
 A _Signer Authorization_ is a message that authorizes a new key pair to generate signatures for a Farcaster account.
 
-When an account is minted, only the custody address can sign messages on its behalf. Users might not want to load this keypair into every device since it increases the risk of account compromise. The custody address, also known as the _Root Signer_, can authorize other keypairs known as _Delegate Signers_. Unlike Root Signers, a Delegate Signer is only allowed to publish off-chain messages and cannot perform any on-chain actions.
+When a fid is minted, only the custody address can sign messages on its behalf. Users might not want to load this keypair into every device since it increases the risk of account compromise. The custody address, also known as the _Root Signer_, can authorize other keypairs known as _Delegate Signers_. Unlike Root Signers, a Delegate Signer is only allowed to publish off-chain messages and cannot perform any on-chain actions.
```
Comment (pfletcherhill) [inline README.md:425]: An fid. There are a couple more cases of this below too to change.

**Review by pfletcherhill — APPROVED** [review]
Approving, but I left a handful of nits and formatting comments. Can you address them before merging?

File: README.md (hunk 11 above)
Comment (varunsrin) [reply README.md:181]: i'm going to do a separate PR to add a commit hook linter

**Pull request merged**